docs/docs/Contributing/contributing-how-to-contribute.md

```diff
@@ -4,157 +4,110 @@ sidebar_position: 1
+2. Add the new remote to your local repository on your local machine:
 
+```bash
+git remote add fork https://github.com/<your_git_username>/langflow.git
+```
 
+### Prepare the development environment
 
-As an open-source project in a rapidly developing field, we are extremely open to contributions, whether in the form of a new feature, improved infra, or better documentation.
+1. Create development hooks.
 
+```bash
+make init
+```
 
+This command sets up the development environment by installing backend and frontend dependencies, building the frontend static files, and initializing the project. It runs `make install_backend`, `make install_frontend`, `make build_frontend`, and finally `uv run langflow run` to start the application.
 
-To contribute to this project, please follow a ["fork and pull request"](https://docs.github.com/en/get-started/quickstart/contributing-to-projects) workflow. Please do not try to push directly to this repo unless you are a maintainer.
+2. Run `make lint`, `make format`, and `make unit_tests` before pushing to the repository.
 
+### Debug
 
-## Local Development {#0388cc3c758d434d994022863a6bafa9}
+The repo includes a `.vscode/launch.json` file for debugging the backend in VSCode, which is a lot faster than debugging with Docker compose. To debug Langflow with the `launch.json` file in VSCode:
 
+1. Open Langflow in VSCode.
+2. Press Ctrl+Shift+D (or Cmd+Shift+D on Mac) to open the Run and Debug view.
+3. Choose a configuration from the dropdown at the top (for example, "Debug Backend").
+4. Click the green play button or press F5 to start debugging.
+5. This allows you to quickly debug different parts of your application, like the backend, frontend, or CLI, directly from VSCode.
 
----
-
-
-You can develop Langflow using docker compose, or locally.
-
-
-We provide a `.vscode/launch.json` file for debugging the backend in VSCode, which is a lot faster than using docker compose.
-
-
-Setting up hooks:
-
-
-`make init`
-
-
-This will install the pre-commit hooks, which will run `make format` on every commit.
-
-
-It is advised to run `make lint` before pushing to the repository.
-
-
-## Run Locally {#5225c2ef0cd6403c9f6c6bbd888115e0}
-
-
----
-
-
-Langflow can run locally by cloning the repository and installing the dependencies. We recommend using a virtual environment to isolate the dependencies from your system.
-
-
-Before you start, make sure you have the following installed:
-
-- Poetry (&gt;=1.4)
-- Node.js
-
-Then, in the root folder, install the dependencies and start the development server for the backend:
+### Run Langflow locally (Poetry and Node.js)
 
+Run Langflow locally by cloning the repository and installing the dependencies. We recommend using a virtual environment like venv or conda to isolate dependencies.
```

Comment:
Are you telling users to run Langflow locally so they can test their changes? It's not clear.

Reply:
Added more explanation